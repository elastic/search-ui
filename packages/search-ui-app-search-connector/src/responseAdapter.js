function adaptation1AdaptFacetValue(facetValue) {
  const { count, value, ...rest } = facetValue;
  return {
    count,
    value: value
      ? value
      : {
          ...rest
        }
  };
}

function adaptation2AddLabelToFacet(fieldName, facet) {
  return {
    field: fieldName,
    ...facet
  };
}

function adaptFacets(facets) {
  if (!facets || Object.keys(facets).length === 0) return facets;

  return Object.entries(facets).reduce((acc, [fieldName, facet]) => {
    const adaptedFacet = facet.map(v => {
      const { type, data, ...rest } = v;
      return adaptation2AddLabelToFacet(fieldName, {
        type,
        data: data.map(adaptation1AdaptFacetValue),
        ...rest
      });
    });

    return {
      ...acc,
      [fieldName]: adaptedFacet
    };
  }, {});
}

export function adaptResponse(response) {
  const facets = response.info.facets;
  const requestId = response.info.meta.request_id;

  const totalPages = response.info.meta.page
    ? response.info.meta.page.total_pages
    : undefined;

  const totalResults = response.info.meta.page
    ? response.info.meta.page.total_results
    : undefined;

  return {
    ...(facets && { facets: adaptFacets(facets) }),
    requestId,
    results: response.rawResults,
    ...(totalPages !== undefined && { totalPages }),
    ...(totalResults !== undefined && { totalResults })
  };
}
