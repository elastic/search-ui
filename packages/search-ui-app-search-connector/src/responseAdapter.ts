function adaptation1AdaptFacetValue(
  facetValue,
  additionalFacetValueFieldsForField = {}
) {
  const hasValue = facetValue.hasOwnProperty("value");
  const { count, value, ...rest } = facetValue;
  return {
    count,
    value: hasValue
      ? value
      : {
          ...rest,
          ...additionalFacetValueFieldsForField
        }
  };
}

function adaptation2AddLabelToFacet(fieldName, facet) {
  return {
    field: fieldName,
    ...facet
  };
}

function adaptFacets(facets, { additionalFacetValueFields = {} }) {
  if (!facets || Object.keys(facets).length === 0) return facets;

  return Object.entries(facets).reduce((acc, [fieldName, facet]) => {
    const adaptedFacet = facet.map(v => {
      const { type, data, ...rest } = v;
      return adaptation2AddLabelToFacet(fieldName, {
        type,
        data: data.map(f =>
          adaptation1AdaptFacetValue(f, additionalFacetValueFields[fieldName])
        ),
        ...rest
      });
    });

    return {
      ...acc,
      [fieldName]: adaptedFacet
    };
  }, {});
}

function limitTo100pages(totalPages) {
  // We limit this to 100 pages since App Search currently cannot page past 100 pages
  return Math.min(totalPages, 100);
}

export function adaptResponse(response, options = {}) {
  const facets = response.info.facets;
  const requestId = response.info.meta.request_id;

  const totalPages =
    response.info.meta.page &&
    typeof response.info.meta.page.total_pages !== "undefined"
      ? limitTo100pages(response.info.meta.page.total_pages)
      : undefined;

  const totalResults = response.info.meta.page
    ? response.info.meta.page.total_results
    : undefined;

  return {
    ...(facets && { facets: adaptFacets(facets, options) }),
    rawResponse: response,
    requestId,
    results: response.rawResults,
    ...(totalPages !== undefined && { totalPages }),
    ...(totalResults !== undefined && { totalResults })
  };
}
