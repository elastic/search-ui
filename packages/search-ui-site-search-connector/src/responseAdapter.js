import { getFacets, getResults } from "./responseAdapters";

export default function adaptResponse(response, documentType) {
  const results = getResults(response.records, documentType);
  const totalPages = response.info[documentType].num_pages;
  const totalResults = response.info[documentType].total_result_count;
  const requestId = "";
  const facets = getFacets(response.info[documentType]);

  return {
    results,
    totalPages,
    totalResults,
    requestId,
    ...(Object.keys(facets).length > 0 && { facets })
  };
}
