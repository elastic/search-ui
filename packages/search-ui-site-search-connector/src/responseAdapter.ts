import { getFacets, getResults } from "./responseAdapters";
import type { SearchResponse } from "./types";

export default function adaptResponse(
  response: SearchResponse,
  documentType: string
) {
  const results = getResults(response.records, documentType);
  const totalPages = response.info[documentType].num_pages;
  const totalResults = response.info[documentType].total_result_count;
  const requestId = "";
  const facets = getFacets(response.info[documentType]);

  return {
    rawResponse: response,
    results,
    totalPages,
    totalResults,
    requestId,
    ...(Object.keys(facets).length > 0 && { facets })
  };
}
