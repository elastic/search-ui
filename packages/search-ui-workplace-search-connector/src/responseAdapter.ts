import type { Facet, FacetValue } from "@elastic/search-ui";

function adaptation1AdaptFacetValue(
  facetValue: FacetValue,
  additionalFacetValueFieldsForField = {}
) {
  const hasValue = Object.prototype.hasOwnProperty.call(facetValue, "value");
  const { count, value, ...rest } = facetValue;
  return {
    count,
    // TODO: Looks like a bug.
    // "value" type is
    // FilterValue | { selected: boolean; }
    // Doesn't look like { selected: boolean; } is the correct return type
    // Also, when testing locally, the "selected" property never appears
    value: hasValue
      ? value
      : {
          ...rest,
          ...additionalFacetValueFieldsForField
        }
  };
}

// Should be facet: Facet, but this results in a type error, see description above
function adaptation2AddLabelToFacet(fieldName: string, facet) {
  return {
    field: fieldName,
    ...facet
  };
}

function adaptFacets(
  facets: Record<string, Facet[]>,
  { additionalFacetValueFields = {} }
) {
  if (!facets || Object.keys(facets).length === 0) return facets;

  return Object.entries(facets).reduce((acc, [fieldName, facet]) => {
    const adaptedFacet = facet.map((v) => {
      const { type, data, ...rest } = v;
      return adaptation2AddLabelToFacet(fieldName, {
        type,
        data: data.map((f) =>
          adaptation1AdaptFacetValue(f, additionalFacetValueFields[fieldName])
        ),
        ...rest
      });
    });

    return {
      ...acc,
      [fieldName]: adaptedFacet
    };
  }, {});
}

function limitTo100pages(totalPages: number): number {
  // We limit this to 100 pages since App Search currently cannot page past 100 pages
  return Math.min(totalPages, 100);
}

export function adaptResponse(response, options = {}) {
  const facets = response.facets;
  const requestId = response.meta.request_id;

  const totalPages =
    response.meta.page && typeof response.meta.page.total_pages !== "undefined"
      ? limitTo100pages(response.meta.page.total_pages)
      : undefined;

  const totalResults = response.meta.page
    ? response.meta.page.total_results
    : undefined;

  return {
    ...(facets && { facets: adaptFacets(facets, options) }),
    rawResponse: response,
    requestId,
    results: response.results,
    ...(totalPages !== undefined && { totalPages }),
    ...(totalResults !== undefined && { totalResults })
  };
}
