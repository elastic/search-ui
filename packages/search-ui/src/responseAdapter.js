function adaptation1AdaptFacetValue(facetValue) {
  const { count, value, ...rest } = facetValue;
  return {
    count,
    value: value
      ? value
      : {
          ...rest
        }
  };
}

function adaptation2AddLabelToFacet(fieldName, facet) {
  return {
    field: fieldName,
    ...facet
  };
}

export function adaptFacets(facets) {
  if (!facets || Object.keys(facets).length === 0) return facets;

  return Object.entries(facets).reduce((acc, [fieldName, facet]) => {
    const adaptedFacet = facet.map(v => {
      const { type, data, ...rest } = v;
      return adaptation2AddLabelToFacet(fieldName, {
        type,
        data: data.map(adaptation1AdaptFacetValue),
        ...rest
      });
    });

    return {
      ...acc,
      [fieldName]: adaptedFacet
    };
  }, {});
}
