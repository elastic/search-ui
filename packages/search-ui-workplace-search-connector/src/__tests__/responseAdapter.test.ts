import { adaptResponse } from "../responseAdapter";
import exampleResponse from "./exampleResponse.json";

const deepClone = (obj) => JSON.parse(JSON.stringify(obj));

const emptyResponse = {
  meta: {
    page: {
      current: 1,
      total_pages: 0,
      total_results: 0,
      size: 5
    },
    warnings: [],
    sources: {
      "621581b6174a804659f9dc16": {
        name: "National parks",
        service_type: "custom"
      }
    },
    request_id: "wVlg15HbT0yyHJSJUlnHQw"
  },
  results: []
};

describe("adaptResponse", () => {
  it("adapts response", () => {
    expect(adaptResponse(exampleResponse)).toMatchSnapshot();
  });

  it("adapts empty response", () => {
    expect(adaptResponse(emptyResponse)).toMatchSnapshot();
  });

  it("adapts facets with empty values", () => {
    const responseWithEmptyFacetValue = deepClone(exampleResponse);
    responseWithEmptyFacetValue.facets.world_heritage_site[0].data[0].value =
      "";
    expect(adaptResponse(responseWithEmptyFacetValue)).toMatchSnapshot();
  });

  it("adapts facets with zero values", () => {
    const responseWithZeroFacetValue = deepClone(exampleResponse);
    responseWithZeroFacetValue.facets.world_heritage_site[0].data[0].value =
      0 as any;
    expect(adaptResponse(responseWithZeroFacetValue)).toMatchSnapshot();
  });

  it("will accept additional facet value fields to inject into response", () => {
    const geoOptions = {
      additionalFacetValueFields: {
        location: {
          center: "73.102, -78.120",
          unit: "u"
        }
      }
    };
    expect(adaptResponse(exampleResponse, geoOptions)).toMatchSnapshot();
  });

  it("will limit total pages to 100", () => {
    const responseWithOver100Pages = deepClone(exampleResponse);
    responseWithOver100Pages.meta.page = {
      current: 1,
      total_pages: 5000,
      total_results: 100000,
      size: 20
    };
    expect(adaptResponse(responseWithOver100Pages)).toMatchSnapshot();
  });
});
