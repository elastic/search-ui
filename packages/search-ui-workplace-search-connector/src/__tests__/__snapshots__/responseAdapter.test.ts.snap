// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adaptResponse adapts empty response 1`] = `
Object {
  "rawResponse": Object {
    "meta": Object {
      "page": Object {
        "current": 1,
        "size": 5,
        "total_pages": 0,
        "total_results": 0,
      },
      "request_id": "wVlg15HbT0yyHJSJUlnHQw",
      "sources": Object {
        "621581b6174a804659f9dc16": Object {
          "name": "National parks",
          "service_type": "custom",
        },
      },
      "warnings": Array [],
    },
    "results": Array [],
  },
  "requestId": "wVlg15HbT0yyHJSJUlnHQw",
  "results": Array [],
  "totalPages": 0,
  "totalResults": 0,
}
`;

exports[`adaptResponse adapts facets with empty values 1`] = `
Object {
  "facets": Object {
    "world_heritage_site": Array [
      Object {
        "data": Array [
          Object {
            "count": 1,
            "value": "",
          },
        ],
        "field": "world_heritage_site",
        "type": "value",
      },
    ],
  },
  "rawResponse": Object {
    "facets": Object {
      "world_heritage_site": Array [
        Object {
          "data": Array [
            Object {
              "count": 1,
              "value": "",
            },
          ],
          "type": "value",
        },
      ],
    },
    "meta": Object {
      "page": Object {
        "current": 1,
        "size": 20,
        "total_pages": 1,
        "total_results": 1,
      },
      "request_id": "W6qPJzEBS0eoUnVG4FZTnw",
      "sources": Object {
        "621581b6174a804659f9dc16": Object {
          "name": "National parks",
          "service_type": "custom",
        },
      },
      "warnings": Array [
        "unknown result field 'image_url'",
      ],
    },
    "results": Array [
      Object {
        "_meta": Object {
          "content_source_id": "621581b6174a804659f9dc16",
          "id": "park_great-smoky-mountains",
          "last_updated": "2022-02-23T00:40:10+00:00",
          "score": 2.057784,
          "source": "custom",
        },
        "acres": Object {
          "raw": 522426.88,
        },
        "content_source_id": Object {
          "raw": "621581b6174a804659f9dc16",
        },
        "date_established": Object {
          "raw": "1934-06-15T05:00:00+00:00",
        },
        "description": Object {
          "snippet": "The Great Smoky Mountains, part of the Appalachian Mountains, span a wide range of elevations",
        },
        "id": Object {
          "raw": "park_great-smoky-mountains",
        },
        "last_updated": Object {
          "raw": "2022-02-23T00:40:10+00:00",
        },
        "location": Object {
          "raw": "35.68,-83.53",
        },
        "nps_link": Object {
          "raw": "https://www.nps.gov/grsm/index.htm",
        },
        "source": Object {
          "raw": "custom",
        },
        "square_km": Object {
          "raw": 2114.2,
        },
        "states": Object {
          "raw": Array [
            "Tennessee",
            "North Carolina",
          ],
        },
        "title": Object {
          "snippet": "Great Smoky Mountains",
        },
        "visitors": Object {
          "raw": 11312786,
        },
        "world_heritage_site": Object {
          "raw": "true",
        },
      },
    ],
  },
  "requestId": "W6qPJzEBS0eoUnVG4FZTnw",
  "results": Array [
    Object {
      "_meta": Object {
        "content_source_id": "621581b6174a804659f9dc16",
        "id": "park_great-smoky-mountains",
        "last_updated": "2022-02-23T00:40:10+00:00",
        "score": 2.057784,
        "source": "custom",
      },
      "acres": Object {
        "raw": 522426.88,
      },
      "content_source_id": Object {
        "raw": "621581b6174a804659f9dc16",
      },
      "date_established": Object {
        "raw": "1934-06-15T05:00:00+00:00",
      },
      "description": Object {
        "snippet": "The Great Smoky Mountains, part of the Appalachian Mountains, span a wide range of elevations",
      },
      "id": Object {
        "raw": "park_great-smoky-mountains",
      },
      "last_updated": Object {
        "raw": "2022-02-23T00:40:10+00:00",
      },
      "location": Object {
        "raw": "35.68,-83.53",
      },
      "nps_link": Object {
        "raw": "https://www.nps.gov/grsm/index.htm",
      },
      "source": Object {
        "raw": "custom",
      },
      "square_km": Object {
        "raw": 2114.2,
      },
      "states": Object {
        "raw": Array [
          "Tennessee",
          "North Carolina",
        ],
      },
      "title": Object {
        "snippet": "Great Smoky Mountains",
      },
      "visitors": Object {
        "raw": 11312786,
      },
      "world_heritage_site": Object {
        "raw": "true",
      },
    },
  ],
  "totalPages": 1,
  "totalResults": 1,
}
`;

exports[`adaptResponse adapts facets with zero values 1`] = `
Object {
  "facets": Object {
    "world_heritage_site": Array [
      Object {
        "data": Array [
          Object {
            "count": 1,
            "value": 0,
          },
        ],
        "field": "world_heritage_site",
        "type": "value",
      },
    ],
  },
  "rawResponse": Object {
    "facets": Object {
      "world_heritage_site": Array [
        Object {
          "data": Array [
            Object {
              "count": 1,
              "value": 0,
            },
          ],
          "type": "value",
        },
      ],
    },
    "meta": Object {
      "page": Object {
        "current": 1,
        "size": 20,
        "total_pages": 1,
        "total_results": 1,
      },
      "request_id": "W6qPJzEBS0eoUnVG4FZTnw",
      "sources": Object {
        "621581b6174a804659f9dc16": Object {
          "name": "National parks",
          "service_type": "custom",
        },
      },
      "warnings": Array [
        "unknown result field 'image_url'",
      ],
    },
    "results": Array [
      Object {
        "_meta": Object {
          "content_source_id": "621581b6174a804659f9dc16",
          "id": "park_great-smoky-mountains",
          "last_updated": "2022-02-23T00:40:10+00:00",
          "score": 2.057784,
          "source": "custom",
        },
        "acres": Object {
          "raw": 522426.88,
        },
        "content_source_id": Object {
          "raw": "621581b6174a804659f9dc16",
        },
        "date_established": Object {
          "raw": "1934-06-15T05:00:00+00:00",
        },
        "description": Object {
          "snippet": "The Great Smoky Mountains, part of the Appalachian Mountains, span a wide range of elevations",
        },
        "id": Object {
          "raw": "park_great-smoky-mountains",
        },
        "last_updated": Object {
          "raw": "2022-02-23T00:40:10+00:00",
        },
        "location": Object {
          "raw": "35.68,-83.53",
        },
        "nps_link": Object {
          "raw": "https://www.nps.gov/grsm/index.htm",
        },
        "source": Object {
          "raw": "custom",
        },
        "square_km": Object {
          "raw": 2114.2,
        },
        "states": Object {
          "raw": Array [
            "Tennessee",
            "North Carolina",
          ],
        },
        "title": Object {
          "snippet": "Great Smoky Mountains",
        },
        "visitors": Object {
          "raw": 11312786,
        },
        "world_heritage_site": Object {
          "raw": "true",
        },
      },
    ],
  },
  "requestId": "W6qPJzEBS0eoUnVG4FZTnw",
  "results": Array [
    Object {
      "_meta": Object {
        "content_source_id": "621581b6174a804659f9dc16",
        "id": "park_great-smoky-mountains",
        "last_updated": "2022-02-23T00:40:10+00:00",
        "score": 2.057784,
        "source": "custom",
      },
      "acres": Object {
        "raw": 522426.88,
      },
      "content_source_id": Object {
        "raw": "621581b6174a804659f9dc16",
      },
      "date_established": Object {
        "raw": "1934-06-15T05:00:00+00:00",
      },
      "description": Object {
        "snippet": "The Great Smoky Mountains, part of the Appalachian Mountains, span a wide range of elevations",
      },
      "id": Object {
        "raw": "park_great-smoky-mountains",
      },
      "last_updated": Object {
        "raw": "2022-02-23T00:40:10+00:00",
      },
      "location": Object {
        "raw": "35.68,-83.53",
      },
      "nps_link": Object {
        "raw": "https://www.nps.gov/grsm/index.htm",
      },
      "source": Object {
        "raw": "custom",
      },
      "square_km": Object {
        "raw": 2114.2,
      },
      "states": Object {
        "raw": Array [
          "Tennessee",
          "North Carolina",
        ],
      },
      "title": Object {
        "snippet": "Great Smoky Mountains",
      },
      "visitors": Object {
        "raw": 11312786,
      },
      "world_heritage_site": Object {
        "raw": "true",
      },
    },
  ],
  "totalPages": 1,
  "totalResults": 1,
}
`;

exports[`adaptResponse adapts response 1`] = `
Object {
  "facets": Object {
    "world_heritage_site": Array [
      Object {
        "data": Array [
          Object {
            "count": 1,
            "value": "true",
          },
        ],
        "field": "world_heritage_site",
        "type": "value",
      },
    ],
  },
  "rawResponse": Object {
    "facets": Object {
      "world_heritage_site": Array [
        Object {
          "data": Array [
            Object {
              "count": 1,
              "value": "true",
            },
          ],
          "type": "value",
        },
      ],
    },
    "meta": Object {
      "page": Object {
        "current": 1,
        "size": 20,
        "total_pages": 1,
        "total_results": 1,
      },
      "request_id": "W6qPJzEBS0eoUnVG4FZTnw",
      "sources": Object {
        "621581b6174a804659f9dc16": Object {
          "name": "National parks",
          "service_type": "custom",
        },
      },
      "warnings": Array [
        "unknown result field 'image_url'",
      ],
    },
    "results": Array [
      Object {
        "_meta": Object {
          "content_source_id": "621581b6174a804659f9dc16",
          "id": "park_great-smoky-mountains",
          "last_updated": "2022-02-23T00:40:10+00:00",
          "score": 2.057784,
          "source": "custom",
        },
        "acres": Object {
          "raw": 522426.88,
        },
        "content_source_id": Object {
          "raw": "621581b6174a804659f9dc16",
        },
        "date_established": Object {
          "raw": "1934-06-15T05:00:00+00:00",
        },
        "description": Object {
          "snippet": "The Great Smoky Mountains, part of the Appalachian Mountains, span a wide range of elevations",
        },
        "id": Object {
          "raw": "park_great-smoky-mountains",
        },
        "last_updated": Object {
          "raw": "2022-02-23T00:40:10+00:00",
        },
        "location": Object {
          "raw": "35.68,-83.53",
        },
        "nps_link": Object {
          "raw": "https://www.nps.gov/grsm/index.htm",
        },
        "source": Object {
          "raw": "custom",
        },
        "square_km": Object {
          "raw": 2114.2,
        },
        "states": Object {
          "raw": Array [
            "Tennessee",
            "North Carolina",
          ],
        },
        "title": Object {
          "snippet": "Great Smoky Mountains",
        },
        "visitors": Object {
          "raw": 11312786,
        },
        "world_heritage_site": Object {
          "raw": "true",
        },
      },
    ],
  },
  "requestId": "W6qPJzEBS0eoUnVG4FZTnw",
  "results": Array [
    Object {
      "_meta": Object {
        "content_source_id": "621581b6174a804659f9dc16",
        "id": "park_great-smoky-mountains",
        "last_updated": "2022-02-23T00:40:10+00:00",
        "score": 2.057784,
        "source": "custom",
      },
      "acres": Object {
        "raw": 522426.88,
      },
      "content_source_id": Object {
        "raw": "621581b6174a804659f9dc16",
      },
      "date_established": Object {
        "raw": "1934-06-15T05:00:00+00:00",
      },
      "description": Object {
        "snippet": "The Great Smoky Mountains, part of the Appalachian Mountains, span a wide range of elevations",
      },
      "id": Object {
        "raw": "park_great-smoky-mountains",
      },
      "last_updated": Object {
        "raw": "2022-02-23T00:40:10+00:00",
      },
      "location": Object {
        "raw": "35.68,-83.53",
      },
      "nps_link": Object {
        "raw": "https://www.nps.gov/grsm/index.htm",
      },
      "source": Object {
        "raw": "custom",
      },
      "square_km": Object {
        "raw": 2114.2,
      },
      "states": Object {
        "raw": Array [
          "Tennessee",
          "North Carolina",
        ],
      },
      "title": Object {
        "snippet": "Great Smoky Mountains",
      },
      "visitors": Object {
        "raw": 11312786,
      },
      "world_heritage_site": Object {
        "raw": "true",
      },
    },
  ],
  "totalPages": 1,
  "totalResults": 1,
}
`;

exports[`adaptResponse will accept additional facet value fields to inject into response 1`] = `
Object {
  "facets": Object {
    "world_heritage_site": Array [
      Object {
        "data": Array [
          Object {
            "count": 1,
            "value": "true",
          },
        ],
        "field": "world_heritage_site",
        "type": "value",
      },
    ],
  },
  "rawResponse": Object {
    "facets": Object {
      "world_heritage_site": Array [
        Object {
          "data": Array [
            Object {
              "count": 1,
              "value": "true",
            },
          ],
          "type": "value",
        },
      ],
    },
    "meta": Object {
      "page": Object {
        "current": 1,
        "size": 20,
        "total_pages": 1,
        "total_results": 1,
      },
      "request_id": "W6qPJzEBS0eoUnVG4FZTnw",
      "sources": Object {
        "621581b6174a804659f9dc16": Object {
          "name": "National parks",
          "service_type": "custom",
        },
      },
      "warnings": Array [
        "unknown result field 'image_url'",
      ],
    },
    "results": Array [
      Object {
        "_meta": Object {
          "content_source_id": "621581b6174a804659f9dc16",
          "id": "park_great-smoky-mountains",
          "last_updated": "2022-02-23T00:40:10+00:00",
          "score": 2.057784,
          "source": "custom",
        },
        "acres": Object {
          "raw": 522426.88,
        },
        "content_source_id": Object {
          "raw": "621581b6174a804659f9dc16",
        },
        "date_established": Object {
          "raw": "1934-06-15T05:00:00+00:00",
        },
        "description": Object {
          "snippet": "The Great Smoky Mountains, part of the Appalachian Mountains, span a wide range of elevations",
        },
        "id": Object {
          "raw": "park_great-smoky-mountains",
        },
        "last_updated": Object {
          "raw": "2022-02-23T00:40:10+00:00",
        },
        "location": Object {
          "raw": "35.68,-83.53",
        },
        "nps_link": Object {
          "raw": "https://www.nps.gov/grsm/index.htm",
        },
        "source": Object {
          "raw": "custom",
        },
        "square_km": Object {
          "raw": 2114.2,
        },
        "states": Object {
          "raw": Array [
            "Tennessee",
            "North Carolina",
          ],
        },
        "title": Object {
          "snippet": "Great Smoky Mountains",
        },
        "visitors": Object {
          "raw": 11312786,
        },
        "world_heritage_site": Object {
          "raw": "true",
        },
      },
    ],
  },
  "requestId": "W6qPJzEBS0eoUnVG4FZTnw",
  "results": Array [
    Object {
      "_meta": Object {
        "content_source_id": "621581b6174a804659f9dc16",
        "id": "park_great-smoky-mountains",
        "last_updated": "2022-02-23T00:40:10+00:00",
        "score": 2.057784,
        "source": "custom",
      },
      "acres": Object {
        "raw": 522426.88,
      },
      "content_source_id": Object {
        "raw": "621581b6174a804659f9dc16",
      },
      "date_established": Object {
        "raw": "1934-06-15T05:00:00+00:00",
      },
      "description": Object {
        "snippet": "The Great Smoky Mountains, part of the Appalachian Mountains, span a wide range of elevations",
      },
      "id": Object {
        "raw": "park_great-smoky-mountains",
      },
      "last_updated": Object {
        "raw": "2022-02-23T00:40:10+00:00",
      },
      "location": Object {
        "raw": "35.68,-83.53",
      },
      "nps_link": Object {
        "raw": "https://www.nps.gov/grsm/index.htm",
      },
      "source": Object {
        "raw": "custom",
      },
      "square_km": Object {
        "raw": 2114.2,
      },
      "states": Object {
        "raw": Array [
          "Tennessee",
          "North Carolina",
        ],
      },
      "title": Object {
        "snippet": "Great Smoky Mountains",
      },
      "visitors": Object {
        "raw": 11312786,
      },
      "world_heritage_site": Object {
        "raw": "true",
      },
    },
  ],
  "totalPages": 1,
  "totalResults": 1,
}
`;

exports[`adaptResponse will limit total pages to 100 1`] = `
Object {
  "facets": Object {
    "world_heritage_site": Array [
      Object {
        "data": Array [
          Object {
            "count": 1,
            "value": "true",
          },
        ],
        "field": "world_heritage_site",
        "type": "value",
      },
    ],
  },
  "rawResponse": Object {
    "facets": Object {
      "world_heritage_site": Array [
        Object {
          "data": Array [
            Object {
              "count": 1,
              "value": "true",
            },
          ],
          "type": "value",
        },
      ],
    },
    "meta": Object {
      "page": Object {
        "current": 1,
        "size": 20,
        "total_pages": 5000,
        "total_results": 100000,
      },
      "request_id": "W6qPJzEBS0eoUnVG4FZTnw",
      "sources": Object {
        "621581b6174a804659f9dc16": Object {
          "name": "National parks",
          "service_type": "custom",
        },
      },
      "warnings": Array [
        "unknown result field 'image_url'",
      ],
    },
    "results": Array [
      Object {
        "_meta": Object {
          "content_source_id": "621581b6174a804659f9dc16",
          "id": "park_great-smoky-mountains",
          "last_updated": "2022-02-23T00:40:10+00:00",
          "score": 2.057784,
          "source": "custom",
        },
        "acres": Object {
          "raw": 522426.88,
        },
        "content_source_id": Object {
          "raw": "621581b6174a804659f9dc16",
        },
        "date_established": Object {
          "raw": "1934-06-15T05:00:00+00:00",
        },
        "description": Object {
          "snippet": "The Great Smoky Mountains, part of the Appalachian Mountains, span a wide range of elevations",
        },
        "id": Object {
          "raw": "park_great-smoky-mountains",
        },
        "last_updated": Object {
          "raw": "2022-02-23T00:40:10+00:00",
        },
        "location": Object {
          "raw": "35.68,-83.53",
        },
        "nps_link": Object {
          "raw": "https://www.nps.gov/grsm/index.htm",
        },
        "source": Object {
          "raw": "custom",
        },
        "square_km": Object {
          "raw": 2114.2,
        },
        "states": Object {
          "raw": Array [
            "Tennessee",
            "North Carolina",
          ],
        },
        "title": Object {
          "snippet": "Great Smoky Mountains",
        },
        "visitors": Object {
          "raw": 11312786,
        },
        "world_heritage_site": Object {
          "raw": "true",
        },
      },
    ],
  },
  "requestId": "W6qPJzEBS0eoUnVG4FZTnw",
  "results": Array [
    Object {
      "_meta": Object {
        "content_source_id": "621581b6174a804659f9dc16",
        "id": "park_great-smoky-mountains",
        "last_updated": "2022-02-23T00:40:10+00:00",
        "score": 2.057784,
        "source": "custom",
      },
      "acres": Object {
        "raw": 522426.88,
      },
      "content_source_id": Object {
        "raw": "621581b6174a804659f9dc16",
      },
      "date_established": Object {
        "raw": "1934-06-15T05:00:00+00:00",
      },
      "description": Object {
        "snippet": "The Great Smoky Mountains, part of the Appalachian Mountains, span a wide range of elevations",
      },
      "id": Object {
        "raw": "park_great-smoky-mountains",
      },
      "last_updated": Object {
        "raw": "2022-02-23T00:40:10+00:00",
      },
      "location": Object {
        "raw": "35.68,-83.53",
      },
      "nps_link": Object {
        "raw": "https://www.nps.gov/grsm/index.htm",
      },
      "source": Object {
        "raw": "custom",
      },
      "square_km": Object {
        "raw": 2114.2,
      },
      "states": Object {
        "raw": Array [
          "Tennessee",
          "North Carolina",
        ],
      },
      "title": Object {
        "snippet": "Great Smoky Mountains",
      },
      "visitors": Object {
        "raw": 11312786,
      },
      "world_heritage_site": Object {
        "raw": "true",
      },
    },
  ],
  "totalPages": 100,
  "totalResults": 100000,
}
`;
